roles/kustomize_deploy/tasks/execute_step.yml

```diff
@@ -5,14 +5,31 @@
       - stage is defined
       - stage_id is defined
       - stage['path'] is defined
-      ## TODO: remove "validations" once architecture repo is up-to-date
-      - (stage.validations is defined and
-         stage.validations | length > 0) or
-        (stage.wait_conditions is defined and
+      - (stage.wait_conditions is defined and
          stage.wait_conditions | length > 0)
       - stage['values'] is defined
       - stage['values'] | length > 0
 
+- name: Assert wait_conditions are correct
```

Comment:
```suggestion
- name: Assert all wait conditions are valid
  vars:
    _stage_name: "stage_{{ stage_id }}"
    _custom_conditions: >-
      {{
        cifmw_architecture_wait_condition[_stage_name] |
        default([])
      }}
    _pattern: '^(oc|kubectl).*wait.*$'
  block:
    - name: Check custom wait conditions
      when: _custom_conditions | length > 0
      ansible.builtin.assert:
        that:
          - item is match(_pattern)
        msg: >-
          The following custom wait condition is not vaild: {{ item }}
        quiet: true
      ignore_errors: true
      register: custom_wait_conditions_results
      loop: "{{ _custom_conditions }}"

    - name: Check builtin wait conditions
      ansible.builtin.assert:
        that:
          - item is match(_pattern)
        msg: >-
          The following builtin wait condition is not vaild: {{ item }}
        quiet: true
      ignore_errors: true
      register: builtin_wait_conditions_results
      loop: "{{ stage.wait_conditions }}"

    - name: Check wait condition validation results
      ansible.builtin.fail:
        msg: "Please review the above failed wait conditions task"
      when: >
        (builtin_wait_conditions_results is failed or
        custom_wait_conditions_results is failed)
```

Reply:
indeed, providing some more data in the error message is a good thing.